purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1241,23 +1249,42 @@ class Purchases internal constructor(
             )
         }
 
-    private fun getProducts(
+    private fun getProductsOfTypes(
+        productIds: Set<String>,
+        types: Set<ProductType>,
+        callback: GetStoreProductsCallback
+    ) = getProductsOfTypes(productIds, types, emptyList(), callback)
+
+    private fun getProductsOfTypes(
         productIds: Set<String>,
-        productType: ProductType,
+        types: Set<ProductType>,
+        collectedStoreProducts: List<StoreProduct>,
         callback: GetStoreProductsCallback
     ) {
-        billing.queryProductDetailsAsync(
-            productType,
-            productIds,
-            { storeProducts ->
-                dispatch {
-                    callback.onReceived(storeProducts)
-                }
-            }, {
-                dispatch {
-                    callback.onError(it)
-                }
-            })
+        val typesRemaining = types.filter { it != ProductType.UNKNOWN }.toMutableSet()
```

Comment:
maybe we can filter out unknown in the call above

Reply:
Ohhh, that's makes so much more sense 😅